packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -1,274 +1,43 @@
-          'Only single filter is available in the MIT version.',
-          '',
-          'You need to upgrade to the XGrid component to unlock this feature.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  }),
-  onRowsScrollEnd: chainPropTypes(PropTypes.any, (props: any) => {
-    if (props.onRowsScrollEnd != null) {
-      return new Error(
-        [
-          `Material-UI: \`onRowsScrollEnd\` is not a valid prop.`,
-          'onRowsScrollEnd is not available in the MIT version.',
-          '',
-          'You need to upgrade to the XGrid component to unlock this feature.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  }),
-  pageSize: chainPropTypes(PropTypes.number, (props: any) => {
-    if (props.pageSize && props.pageSize > MAX_PAGE_SIZE) {
-      return new Error(
-        [
-          `Material-UI: \`<DataGrid pageSize={${props.pageSize}} />\` is not a valid prop.`,
-          `Only page size below ${MAX_PAGE_SIZE} is available in the MIT version.`,
-          '',
-          'You need to upgrade to the XGrid component to unlock this feature.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  }),
-  pagination: (props: any) => {
-    if (props.pagination === false) {
-      return new Error(
-        [
-          'Material-UI: `<DataGrid pagination={false} />` is not a valid prop.',
-          'Infinite scrolling is not available in the MIT version.',
-          '',
-          'You need to upgrade to the XGrid component to disable the pagination.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  },
-  rows: PropTypes.array.isRequired,
-  scrollEndThreshold: chainPropTypes(PropTypes.number, (props: any) => {
-    if (props.scrollEndThreshold) {
-      return new Error(
-        [
-          `Material-UI: \`<DataGrid scrollEndThreshold={${props.scrollEndThreshold}} />\` is not a valid prop.`,
-          'scrollEndThreshold is not available in the MIT version.',
-          '',
-          'You need to upgrade to the XGrid component to unlock this feature.',
-        ].join('\n'),
-      );
-    }
-    return null;
-  }),
-  selectionModel: chainPropTypes(
-    PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.array]),
-    (props: any) => {
-      if (
-        !props.checkboxSelection &&
-        Array.isArray(props.selectionModel) &&
-        props.selectionModel.length > 1
-      ) {
-        return new Error(
-          [
-            `Material-UI: \`<DataGrid selectionModel={${JSON.stringify(
-              props.selectionModel,
-            )}} />\` is not a valid prop.`,
-            'selectionModel can only contain 1 item in DataGrid without checkbox selection.',
-            '',
-            'You need to upgrade to the XGrid component to unlock multiple selection.',
-          ].join('\n'),
-        );
-      }
-      return null;
-    },
-  ),
-} as any;
+DataGrid.propTypes = DATA_GRID_PROPTYPES;
```

Comment:
AFAIK, we can't isolate the prop types in a different file. It prevents https://www.npmjs.com/package/babel-plugin-transform-react-remove-prop-types to remove them in production. Also, in v5 they will be automatically generated anyway.